### ISSUE TITLE
Documentation: How to add a ROS node to Apollo.

### ISSUE THREAD

**Issue by draturi** [ApolloAuto/apollo#3664]
I would like add a ROS module to apollo (similar to how apollo comes with usb_cam ros node). Do apollo developers have a guide to show how to do this?

In addition is there a how-to doc on how to create an Apollo-specific node? (a node the uses the adapter manager)

**Comment by draturi** [comment]
Addition: How do you add ros dependencies to a docker container (so that they show up in /home/tmp/share)

**Comment by draturi** [comment]
Okay thank you.

How about installing a ROS indigo package?

**Comment by lianglia-apollo** [comment]
I think you can install the package inside docker and commit a new docker.
For ros installation, please refer http://wiki.ros.org/ROS/Installation.
For docker commit, please refer https://docs.docker.com/engine/reference/commandline/commit/.

**Comment by draturi** [comment]
I tried to install the package using apt-get, but the package got installed in /opt/ros/indigo and not in /home/tmp/ros/. This is where the problem lies, because /opt/ros/indigo is not on the ROS_PACKAGE_PATH. The apollo specific ROS installation is at /home/tmp/ros. Have you run into this before?

Once this works i will commit a new docker image.

**Comment by quning78** [comment]
you have to rebuild the package based on apollo-platform, otherwise, it could be in-compatible with Apollo. @draturi

**Comment by draturi** [comment]
Solved this: The procedure I used:
1. install the 3rd party dep (diagnostic_updater in this case) using `apt-get` in docker
2. Compared `/home/tmp/ros` and `/opt/ros/indigo` and copied over the newly installed files from `/opt...` to `/home... ` 
3. Add a build directive in apollo.sh for the source i was originally trying to build.
4. Run `./apollo.sh build_new_package`

this way i was able to build the ROS indigo package that I originally wanted to build.

Thanks @quning78 @lianglia-apollo for you help.
